rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1661,6 +1661,173 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
 		and packageRef count = openTradeNoExecutionDetails count
 		and packageRef distinct count = 1
 
+func Create_RepricePrimitiveInstruction: <"Creates the primitive instructions for a repricing that alters the collateral quantity and value of the the trade. Traction value and variation margin are processed separately as are transfers of cash and securities.">
+	inputs:
+		tradeState TradeState (1..1) <"The original trade to be repriced.">
+		newAllinPrice number(1..1) <"The collateral all-in price.">
+		newAssetQuantity number(1..1) <"The collateral quantity.">
```

Comment:
After further review this may not be necessary (as we touched on in our call). If you want to leave these mandatory then I don't see this is as an issue.